cmd/virt-handler/virt-handler.go

```diff
@@ -103,9 +111,11 @@ func main() {
 		panic(err)
 	}
 
+	configDiskClient := configdisk.NewConfigDiskClient()
```

Comment:
On controller-startup, after all chaches are synchronized, we have to clean the disks for all VMs which are in a final state or which have disappeared form the cluster. It should be possible to create a "Reference" to a VM (a VM object only with name and namespace set), based on the directories. Then `disks.Undefine(vmref)` can be invoked for every one of it. 

We would clean the volumes for all VMs which:

 * Are in a final state or removed from the cluster, and
 * Are not currently running on the node (since we are in the synchronization phase, we want to avoid deleting volumes of not yet stopped domains)

Reply:
Another good catch!

It should be enough to remove directories for any VM that isn't in the cluster still. VMs still known by libvirt in any state will get cleaned up eventually.